dataclasses_json/core.py

```diff
@@ -244,7 +244,16 @@ def _support_extended_types(field_type, field_value):
         res = (field_value
                if isinstance(field_value, UUID)
                else UUID(field_value))
-    elif _issubclass_safe(field_type, (int, float, str, bool)):
+    elif _issubclass_safe(field_type, bool):
+        # issubclass(bool, int) -> True
+        # thus values >1 will always convert to True in the next clause, unless intercepted
+        if field_value in ('True', 'true', True, 1):
+            res = True
+        elif field_value in ('False', 'false', False, 0):
+            res = False
+        else:
+            raise ValueError(f"Value {field_value} of input type {field_type.__name__} cannot be decoded as boolean")
+    elif _issubclass_safe(field_type, int) or _issubclass_safe(field_type, float) or _issubclass_safe(field_type, str):
```

Comment:
Please keep  the tuple notation here

Reply:
This is not correct because that method doesn't accept tuple. It was an error in original implementation

Reply:
The argument is passed directly to `issubclass` which accepts a tuple.
https://python-reference.readthedocs.io/en/latest/docs/functions/issubclass.html

Reply:
I see, however, there is one twist there: https://docs.python.org/3.11/library/functions.html#issubclass

> Changed in version 3.10: classinfo can be a [Union Type](https://docs.python.org/3.11/library/stdtypes.html#types-union).

I'll see if this affects anything